client/src/Try/API.purs

```diff
@@ -123,31 +125,23 @@ instance decodeCompileResult :: Decode CompileResult where
     CompileSuccess <$> genericDecode decodingOptions f
     <|> CompileFailed <$> genericDecode decodingOptions f
 
-foreign import get_
-  :: EffectFn3
-            String
-            (EffectFn1 String Unit)
-            (EffectFn1 String Unit)
-            Unit
-
--- | A wrapper for `get` which uses `ContT`.
-get :: String -> ExceptT String (ContT Unit Effect) String
-get uri = ExceptT (ContT \k -> runEffectFn3 get_ uri (mkEffectFn1 (k <<< Right)) (mkEffectFn1 (k <<< Left)))
-
--- | POST the specified code to the Try PureScript API, and wait for
--- | a response.
-foreign import compile_
-  :: EffectFn4
-            String
-            String
-            (EffectFn1 Foreign Unit)
-            (EffectFn1 String Unit)
-            Unit
-
--- | A wrapper for `compileApi` which uses `ContT`.
-compile
-  :: String
-  -> String
-  -> ExceptT String (ContT Unit Effect)
-       (Either (NonEmptyList ForeignError) CompileResult)
-compile endpoint code = ExceptT (ContT \k -> runEffectFn4 compile_ endpoint code (mkEffectFn1 (k <<< Right <<< runExcept <<< decode)) (mkEffectFn1 (k <<< Left)))
+get :: URL -> ExceptT String Aff String
+get url = ExceptT $ AX.get AXRF.string url >>= case _ of
+  Left e ->
+    pure $ Left $ printError e
+  Right { status } | status >= StatusCode 400 ->
+    pure $ Left $ "Received error status code: " <> show status
+  Right { body } ->
+    pure $ Right body
```

Comment:
I believe the second check (which guards against status codes over 400) is technically unnecessary, but I couldn't verify that at a quick breeze through the Affjax documentation so I've included it here. I can remove these checks if they're redundant, though.

This doesn't really need to be in `ExceptT`, but doing so allows me to minimize the diff everywhere else.

Reply:
Can you check and see what happens if you get a 404 status code? I haven't been able to verify from a quick glance through the documentation either but I actually suspect it's not redundant. Either way, I think we should know the answer before merging.

Reply:
I’ll verify this check is necessary. Do you have any opinion on the code as-is if the check does turn out to be necessary?

Reply:
The code seems fine to me as-is if the check is necessary. If it's not necessary, I think we should remove it.

Reply:
@hdgarrood Yep, it turns out this check is necessary.

If the request is made with a malformed URL:
<img width="1144" alt="Screen Shot 2020-12-15 at 6 14 49 PM" src="https://user-images.githubusercontent.com/10245104/102296321-f30a7c80-3f01-11eb-957e-5259b66fc08b.png">

If the request returns a 404:
<img width="380" alt="Screen Shot 2020-12-15 at 6 17 01 PM" src="https://user-images.githubusercontent.com/10245104/102296314-f00f8c00-3f01-11eb-8fed-b611dc990b5a.png">